.github/workflows/build_and_test.yml

```diff
@@ -821,3 +821,27 @@ jobs:
       with:
         name: unit-tests-log-docker-integration--8-${{ needs.configure-jobs.outputs.hadoop }}-hive2.3
         path: "**/target/unit-tests.log"
+
+  shellcheck:
+    needs: [configure-jobs, precondition]
+    name: Run shell code style check
+    if: needs.configure-jobs.outputs.type == 'regular' && fromJson(needs.precondition.outputs.required).build == 'true'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Spark repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          repository: apache/spark
+          ref: master
+      - name: Sync the current branch with the latest in Apache Spark
+        if: github.repository != 'apache/spark'
+        run: |
+          git fetch https://github.com/$GITHUB_REPOSITORY.git ${GITHUB_REF#refs/heads/}
+          git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' merge --no-commit --progress --squash FETCH_HEAD
+          git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' commit -m "Merged commit"
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```

Comment:
quick question, can we add a dev script? like dev/linter-python, dev/linter-scala

Reply:
Sure, I will add a `dev/linter-shell` to do this.